rviz_common/src/rviz_common/load_resource.cpp

```diff
@@ -30,65 +30,65 @@
 
 #include "rviz_common/load_resource.hpp"
 
-// #include <boost/filesystem.hpp>
-// #include <ros/package.h>
-// #include <ros/ros.h>
+#include <string>
 
-#include <QPixmapCache>
-#include <QPainter>
+#include <QFile>  // NOLINT: cpplint cannot handle the include order here
+#include <QPainter>  // NOLINT: cpplint cannot handle the include order here
+#include <QPixmapCache>  // NOLINT: cpplint cannot handle the include order here
+
+#include "ament_index_cpp/get_package_share_directory.hpp"
+#include "ament_index_cpp/get_package_prefix.hpp"
 
 #include "rviz_common/logging.hpp"
 
 namespace rviz_common
 {
 
-// TODO(wjwwood): replace this with ROS 2 and boost-free version
-// boost::filesystem::path getPath(QString url)
-// {
-//   boost::filesystem::path path;
-
-//   if (url.indexOf("package://", 0, Qt::CaseInsensitive) == 0) {
-//     QString package_name = url.section('/', 2, 2);
-//     QString file_name = url.section('/', 3);
-//     path = ros::package::getPath(package_name.toStdString());
-//     path = path / file_name.toStdString();
-//   } else if (url.indexOf("file://", 0, Qt::CaseInsensitive) == 0) {
-//     path = url.section('/', 2).toStdString();
-//   } else {
-//     ROS_ERROR("Invalid or unsupported URL: '%s'", url.toStdString().c_str() );
-//   }
-
-//   return path;
-// }
-
+QString getFilePath(const QString & url)
+{
+  QString file_path = "";
+  if (url.indexOf("package://", 0, Qt::CaseInsensitive) == 0) {
+    QString package_name = url.section('/', 2, 2);
+    QString file_name = url.section('/', 3);
+    try {
+      std::string package_prefix =
+        ament_index_cpp::get_package_share_directory(package_name.toStdString());
+      file_path = QString::fromStdString(package_prefix) + "/" + file_name;
+    } catch (const ament_index_cpp::PackageNotFoundError & error) {
+      RVIZ_COMMON_LOG_ERROR_STREAM("Package not found: " << error.what());
+    }
+  } else if (url.indexOf("file://", 0, Qt::CaseInsensitive) == 0) {
+    file_path = url.section('/', 2);
+  } else {
+    RVIZ_COMMON_LOG_ERROR("Invalid or unsupported URL: " + url.toStdString());
+  }
+  return file_path;
+}
```

Comment:
Is there a reason you're doing this manually with `ament_index_cpp` and not with `resource_retriever`?

For example, this code uses `resource_retriever` to do basically the same thing:

https://github.com/ros2/rviz/blob/222302786be38c5c0b173c53cf08f25ae32c2e11/rviz_rendering/src/rviz_rendering/mesh_loader.cpp#L84-L96

That's in service of loading mesh files so they can be interpreted by assimp or the STL parser.

Reply:
The reason was that I didn't see that `QPixmap` offered a possibility to load data in the form of a `uchar` array, which is what we're given by `resource_retriever`. In addition, I thought that it would then be necessary to distinguish between file formats. Turns out, neither is the case, so I switched to using the resource_retriever.

It would be nice though if the `resource_retriever` had a method to just give us the full path to the resource. That, however, isn't possible.